src/Microsoft.OData.Core/UrlValidation/ValidationEngine/PathSegmentValidator.cs

```diff
@@ -164,10 +164,13 @@ public override void Handle(OperationImportSegment segment)
                 ValidateItem(parameter);
             }
 
-            foreach (IEdmOperation operation in segment.OperationImports)
+            foreach (IEdmOperationImport operation in segment.OperationImports)
             {
                 ValidateItem(operation);
-                ValidateItem(operation.ReturnType);
+                if (operation.EntitySet != null)
+                {
+                    ValidateItem(operation.EntitySet);

```

Comment:
What if the return type is not from an entity set?

Reply:
Curious `operation.EntitySet` was preferred over `operation.Operation.ReturnType` which seems closer to what the original code intended.

@xuzhg any thoughts on this?